CIME/XML/env_batch.py

```diff
@@ -224,11 +224,33 @@ def get_job_overrides(self, job, case):
             if thread_count:
                 overrides["thread_count"] = thread_count
         else:
-            total_tasks = case.get_value("TOTALPES") * int(case.thread_count)
+            total_tasks = case.get_value("TOTALPES")
```

Comment:
This change scares me a lot. Looking at a couple lines below, I see total_tasks being multiplied by thread_count. It makes no sense how that ever worked because that would have the thread_count multiplied twice. So I would approve of the change if that was the only use of total_tasks, but it isn't.

Reply:
I removed it because I the behavior I saw was that total_tasks was totalpes*thread_count*thread_count (twice).  With this removed, total_tasks=totalpes*thread_count which I think is the intent.

Reply:
Ah gotcha so this was just a bug then. So `task_count` from a jobs override is equivalent to `TOTALPES`?

Reply:
Yes - that's correct.

Reply:
@jedwards4b , I do see the multiply below on line 229 is definitely wrong if we keep the original code for line 227. Are you saying case.get_value("TOTALPES") already takes threads into account?

Reply:
If `task_count` and `TOTALPES` are equivalent then this makes more sense.
```python
            total_tasks = case.get_value("TOTALPES")
            thread_count = case.thread_count
        
        total_tasks *= thread_count
        
        if int(total_tasks) < case.get_value("MAX_TASKS_PER_NODE"):
            overrides["max_tasks_per_node"] = int(total_tasks)
```